### ISSUE TITLE
Card typings for values

### ISSUE THREAD

**Issue by TejasQ** [contiamo/operational-ui#629]
Currently the Card component has a really nice functionality that will allow you to give it `data`, and then access/filter/format the `data` in your card as needed.

## Demo

![08b6fafb-e2c1-427d-9f1d-f66d8c3d86f0](https://user-images.githubusercontent.com/1761469/43310914-38388c44-9189-11e8-8e50-275fbd1ef707.gif)

## [Example](http://operational-ui.netlify.com/#!/Card/5)

[The typing for this](https://github.com/contiamo/operational-ui/pull/627/files#diff-a35de8bc4180536d7c718152b396c178R48) is a little bit sketchy and currently does not satisfy the typescript compiler.

This issue aims to adjust this typing.

**Comment by stereobooster** [comment]
```ts
type A<T extends { [key: string]: any }> =
    { [P in keyof T]?: (value: T[P]) => null };
let x: A<{ a: string, b: number, c: symbol }>;
x.a("a")
x.b(1)
x.c(Symbol(1))
```

**Comment by TejasQ** [comment]
Thanks, @stereobooster! Want to submit a PR?

**Comment by TejasQ** [comment]
Fixed in #655
